test/e2e/suites/apivalidations/openstackmachine_test.go

```diff
@@ -63,6 +63,14 @@ var _ = Describe("OpenStackMachine API validations", func() {
 		By("Modifying the providerID")
 		machine.Spec.ProviderID = ptr.To("bar")
 		Expect(k8sClient.Update(ctx, machine)).NotTo(Succeed(), "Updating providerID should fail")
+
+		By("Setting the identityRef")
+		machine.Spec.IdentityRef = ptr.To(infrav1.OpenStackIdentityReference{Name: "foo", CloudName: "GCP"})
+		Expect(k8sClient.Update(ctx, machine)).To(Succeed(), "Setting identityRef should succeed")
+
+		// By("Modifying the identityRef")
+		// machine.Spec.IdentityRef = ptr.To(infrav1.OpenStackIdentityReference{Name: "bar", CloudName: "AWS"})
+		// Expect(k8sClient.Update(ctx, machine)).To(Succeed(), "Updating identityRef should succeed")
```

Comment:
Is it this one that fails?

Reply:
I have not uncommented it yet because it fails before.

If I change `providerID` and `identityRef` in different tests, it works fine. If I change both values withing the same test, if fails. I updated `cluster-api-provider-openstack/pkg/webhooks/openstackmachine_webhook.go` - `ValidateUpdate()` a bit to print old and new objects on fail. Here the results of the fail are:
```
Message: "admission webhook \"validation.openstackmachine.infrastructure.cluster.x-k8s.io\" denied the request: OpenStackMachine.infrastructure.cluster.x-k8s.io \"machine-vjrtn\" is invalid: spec: Forbidden: cannot be modified

old:map[flavor: image:map[filter:map[name:test-image]] providerID:foo]

new:map[flavor: image:map[filter:map[name:test-image]] providerID:bar]"
```
Please clarify if it is expected.

Reply:
OK, it looks like I got the reason for the failure when updating both `providerID` and `identityRef`. The test failed because updating `providerID` for the second time failed. Once I removed it, the test with updating `providerID` once and updating `identityRef` twice succeeded. Please clarify whether this test with updating both `providerID` and `identityRef` should be kept or it should be deleted